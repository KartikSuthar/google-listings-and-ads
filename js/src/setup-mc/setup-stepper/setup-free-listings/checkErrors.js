/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

const checkErrors = (
	values,
	shippingRates,
	shippingTimes,
	finalCountryCodes
) => {
	const errors = {};

	/**
	 * Check shipping rates.
	 */
	if ( ! values.shipping_rate ) {
		errors.shipping_rate = __(
			'Please select a shipping rate option.',
			'google-listings-and-ads'
		);
	}

	if (
		values.shipping_rate === 'flat' &&
		shippingRates.length < finalCountryCodes.length
	) {
		errors.shipping_rate = __(
			'Please specify shipping rates for all the countries.',
			'google-listings-and-ads'
		);
	}

	if (
		values.shipping_rate === 'flat' &&
		values.offers_free_shipping &&
		! values.free_shipping_threshold
	) {
		errors.free_shipping_threshold = __(
			'Please specify minimum order price for free shipping',
			'google-listings-and-ads'
		);
	}

	/**
	 * Check shipping times.
	 */
	if ( ! values.shipping_time ) {
		errors.shipping_time = __(
			'Please select a shipping time option.',
			'google-listings-and-ads'
		);
	}

	if (
		values.shipping_time === 'flat' &&
		shippingTimes.length < finalCountryCodes.length
	) {
		errors.shipping_time = __(
			'Please specify shipping times for all the countries.',
			'google-listings-and-ads'
		);
	}

	/**
	 * Check tax rate (required for U.S. only).
	 */
	if ( finalCountryCodes.includes( 'US' ) && ! values.tax_rate ) {
		errors.tax_rate = __(
			'Please specify tax rate option.',
			'google-listings-and-ads'
		);
	}

	/**
	 * Pre-launch checklist.
	 */
	if ( ! values.website_live ) {
		errors.website_live = __(
			'Please check the requirement.',
			'google-listings-and-ads'
		);
	}

	if ( ! values.checkout_process_secure ) {
		errors.checkout_process_secure = __(
			'Please check the requirement.',
			'google-listings-and-ads'
		);
	}

	if ( ! values.payment_methods_visible ) {
		errors.payment_methods_visible = __(
			'Please check the requirement.',
			'google-listings-and-ads'
		);
	}

	if ( ! values.refund_tos_visible ) {
		errors.refund_tos_visible = __(
			'Please check the requirement.',
			'google-listings-and-ads'
		);
	}

	if ( ! values.contact_info_visible ) {
		errors.contact_info_visible = __(
			'Please check the requirement.',
			'google-listings-and-ads'
		);
	}

	return errors;
};

export default checkErrors;
