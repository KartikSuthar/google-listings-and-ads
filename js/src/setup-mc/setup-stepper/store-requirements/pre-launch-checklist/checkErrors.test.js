/**
 * Internal dependencies
 */
import checkErrors from './checkErrors';

/**
 * checkErrors function returns an object, and if any checks are not passed,
 * set properties respectively with an error message to indicate it.
 *
 * To test each cases separately, the meaning of "should (not) pass" in test descriptions are:
 * should pass - returned object should not contain respective property.
 * should not pass - returned object should contain respective property with an error message.
 */
describe( 'checkErrors', () => {
	it( 'When all checks are passed, should return an empty object', () => {
		const values = {
			website_live: true,
			checkout_process_secure: true,
			payment_methods_visible: true,
			refund_tos_visible: true,
			contact_info_visible: true,
		};

		const errors = checkErrors( values );

		expect( errors ).toStrictEqual( {} );
	} );

	it( 'should indicate multiple unpassed checks by setting properties in the returned object', () => {
		const errors = checkErrors( {} );

		expect( errors ).toHaveProperty( 'website_live' );
		expect( errors ).toHaveProperty( 'checkout_process_secure' );
	} );

	describe( 'Requirements', () => {
		it( 'When there are any requirements are not true, should not pass', () => {
			// Not set yet
			let errors = checkErrors( {}, [], [], [] );

			expect( errors ).toHaveProperty( 'website_live' );
			expect( errors.website_live ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'checkout_process_secure' );
			expect( errors.checkout_process_secure ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'payment_methods_visible' );
			expect( errors.payment_methods_visible ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'refund_tos_visible' );
			expect( errors.refund_tos_visible ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'contact_info_visible' );
			expect( errors.contact_info_visible ).toMatchSnapshot();

			// Invalid value
			const values = {
				website_live: false,
				checkout_process_secure: 1,
				payment_methods_visible: 'true',
				refund_tos_visible: [],
				contact_info_visible: {},
			};

			errors = checkErrors( values, [], [], [] );

			expect( errors ).toHaveProperty( 'website_live' );
			expect( errors.website_live ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'checkout_process_secure' );
			expect( errors.checkout_process_secure ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'payment_methods_visible' );
			expect( errors.payment_methods_visible ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'refund_tos_visible' );
			expect( errors.refund_tos_visible ).toMatchSnapshot();

			expect( errors ).toHaveProperty( 'contact_info_visible' );
			expect( errors.contact_info_visible ).toMatchSnapshot();
		} );

		it( 'When all requirements are true, should pass', () => {
			const values = {
				website_live: true,
				checkout_process_secure: true,
				payment_methods_visible: true,
				refund_tos_visible: true,
				contact_info_visible: true,
			};

			const errors = checkErrors( values, [], [], [] );

			expect( errors ).not.toHaveProperty( 'website_live' );
			expect( errors ).not.toHaveProperty( 'checkout_process_secure' );
			expect( errors ).not.toHaveProperty( 'payment_methods_visible' );
			expect( errors ).not.toHaveProperty( 'refund_tos_visible' );
			expect( errors ).not.toHaveProperty( 'contact_info_visible' );
		} );
	} );
} );
