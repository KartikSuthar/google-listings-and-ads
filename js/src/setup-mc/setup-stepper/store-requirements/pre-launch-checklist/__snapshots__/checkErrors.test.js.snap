// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 1`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 2`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 3`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 4`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 5`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 6`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 7`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 8`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 9`] = `"Please check the requirement."`;

exports[`checkErrors Requirements When there are any requirements are not true, should not pass 10`] = `"Please check the requirement."`;
